eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/ExtendedBitcoinClient.scala

```diff
@@ -34,81 +34,81 @@ class ExtendedBitcoinClient(val rpcClient: BitcoinJsonRPCClient) {
 
   implicit val formats = org.json4s.DefaultFormats
 
-  def getTxConfirmations(txId: String)(implicit ec: ExecutionContext): Future[Option[Int]] =
-    rpcClient.invoke("getrawtransaction", txId, 1) // we choose verbose output to get the number of confirmations
+  def getTxConfirmations(txid: ByteVector32)(implicit ec: ExecutionContext): Future[Option[Int]] =
+    rpcClient.invoke("getrawtransaction", txid.toHex, 1) // we choose verbose output to get the number of confirmations
```

Comment:
Instead of doing `txid.toHex`, can't you do the same as `BasicBitcoinJsonRPCClient` and add a serializer for ByteVector32? Maybe worth adding a `Serializers.scala` file in the `rpc` folder to share that.

Reply:
Since the serializer is declared in `BasicBitcoinJsonRPCClient` we can already pass `ByteVector32` to 'invoke', i applied the changes in the latest commit.